lib/active_model/serializer/attributes.rb

```diff
@@ -55,6 +55,9 @@ def attributes(*attrs)
         #     end
         def attribute(attr, options = {}, &block)
           key = options.fetch(:key, attr)
+          # replace ? at the end of an attribute, as JSON API disallows the use
+          # of ? at the end of keys.
+          key = key.to_s.gsub(/\?\z/, '').to_sym
```

Comment:
I didn't know sub could handle regex -- that's why I use it. 

http://ruby-doc.org/core-2.2.0/String.html#method-i-sub
well, whaddayaknow.

oh, it looks like gsub replaces all occurrences, and sub replaces just the first occurrence.

that leads me to believe sub would be slightly faster. I'll swap it out.